File: Sample Algorithms/Exam Study/Recursion Exercies/5.py
# Write a recursive function that replaces the small characters with large characters

def replaceLowercase(array,n,uparr):
    if (n > len(array)-1):
        return
    elif (ord('a') <= ord(array[n]) <= ord('z')):
        uparr.append(chr(ord(array[n]) -32))
    else:
        uparr.append(array[n])
    replaceLowercase(array,n+1,uparr)


a = "This IS suPPosed to be UppEr CasE"
b = []
replaceLowercase(a,0,b)
print(a)
print(''.join(b))

File: Sample Algorithms/Exam Study/Recursion Exercies/5.c
// Write a recursive function that replaces the small characters with large characters


#include <stdio.h>
#include <string.h>
#define CONVERT 32

void replaceLowercase(char* arr,int n){
    if(n < 0){
        return;
    }
    if(arr[n] >= 97 && arr[n] <=122){
        arr[n] -= CONVERT;
    } 
    replaceLowercase(arr,n-1);
}

void main() {
    char array[] = "HeLlo worLd";
    int end = strlen(array)-1;
    replaceLowercase(array,end);
    printf("new string is: %s",array);
}